generators/chipyard/src/main/scala/harness/HarnessBinders.scala

```diff
@@ -205,40 +203,57 @@ class WithTiedOffDMI extends HarnessBinder({
   }
 })
 
-class WithSerialTLTiedOff extends HarnessBinder({
-  case (th: HasHarnessInstantiators, port: SerialTLPort) => {
-    if (DataMirror.directionOf(port.io.clock) == Direction.Input) {
-      port.io.clock := false.B.asClock
+// If tieoffs is specified, a list of serial portIds to tie off
+// If tieoffs is unspecified, ties off all serial ports
+class WithSerialTLTiedOff(tieoffs: Option[Seq[Int]] = None) extends HarnessBinder({
+  case (th: HasHarnessInstantiators, port: SerialTLPort, chipId: Int) if (tieoffs.map(_.contains(port.portId)).getOrElse(true)) => {
+    port.io match {
+      case io: DecoupledSerialIO => io.out.ready := false.B; io.in.valid := false.B; io.in.bits := DontCare;
+      case io: SourceSyncSerialIO => {
+        io.clock_in := false.B.asClock
+        io.reset_in := false.B.asAsyncReset
+        io.in := DontCare
+        io.credit_in := DontCare
+      }
+    }
+    port.io match {
+      case io: InternalSyncSerialIO =>
+      case io: ExternalSyncSerialIO => io.clock_in := false.B.asClock
+      case _ =>
     }
-    port.io.bits.out.ready := false.B
-    port.io.bits.in.valid := false.B
-    port.io.bits.in.bits := DontCare
   }
 })
 
 class WithSimTSIOverSerialTL extends HarnessBinder({
-  case (th: HasHarnessInstantiators, port: SerialTLPort) => {
-    val bits = port.io.bits
-    if (DataMirror.directionOf(port.io.clock) == Direction.Input) {
-      port.io.clock := th.harnessBinderClock
+  case (th: HasHarnessInstantiators, port: SerialTLPort, chipId: Int) if (port.portId == 0) => {
```

Comment:
Does the SerialTL port to external memory / bringup FPGA have to be on port 0? If so, is this documented somewhere?

Reply:
No, its not required to be port0, but for convention, I think it should be. This HarnessBinder assumes its port0, as you've noticed. We should stick to this standard, but I'm not sure this is easy to enforce in the generator (since the generator doesn't know what a serial-TL port will connect to.).

Reply:
That seems reasonable. It does seem difficult to add an assert for that. How about adding a config to the TL-serdesser which allows for adding a port instead of requiring the user to configure all the ports at once? Like WithSerialTL but instead of passing a seq, pass SerialTLParams and (optionally?) an index. Since the abstract config already adds the external serial-TL port, this will help enforce that port0 is the external port and overwriting all the ports can be use-at-your-own-risk.
This could probably be added later though. I do think the chipyard documentation should be updated at some point with all the chiplet/multi-chiptop changes, and the external port0 thing would be good to note.

Reply:
> How about adding a config to the TL-serdesser which allows for adding a port instead of requiring the user to configure all the ports at once? 

Yeah, that can be added. I think it will inevitably be confusing, however, to keep track of what is being "appended to", vs "overwritten".